### DISCUSSION TITLE
VerifierSettings .OmitContentFromException() and .OnVerifyMismatch() that are not global

### DISCUSSION THREAD

**Discussion by SeanFeldman** [VerifyTests/Verify#1000] [General]
`VerifierSettings.OmitContentFromException()` and `VerifierSettings.OnVerifyMismatch()` were applied on all Verify tests.

Is there an option to scope these settings per test case rather than all tests in the project?

**Comment by SimonCropp** [comment]
use case?

**Comment by SeanFeldman** [comment]
A fairly large verified and received output that doesn't help to inspect visually w/o running locally.
When there's a mismatch between the two, it's usually due to code added or removed (NSB handlers) and the instructions to tell developers what needs to happen. In short, when running, outputting anything other than instructions how to fix the mismatch doesn't help but it needs to be on a test level and not for a whole test project.

My workaround was to override the comparer but I'd rather do it in a way that is more consistent with the global API.

```
        await Verify(lookup)
            .UseDirectory("ApprovalFiles")
            .UseStringComparer((_, _, _) =>
                Task.FromResult(CompareResult.NotEqual(
                    """
                        Contracts handled by the endpoints verified and received are not matching.\n
                        1. Check all events are subscribed to and validate Infrastructure project is reflecting the changes.\n
                        2. Inspect routing information for no longer dispatched and handled commands.\n
                    """
                )));
```

**Reply by SimonCropp** [reply]
> A fairly large verified and received output that doesn't help to inspect visually w/o running locally.

wouldnt that be better handled with Verify.DiffPlex in compact mode?

https://github.com/VerifyTests/Verify.DiffPlex#output-types

**Comment by SeanFeldman** [comment]
`Verify.DiffPlex` is a good suggestion and it would potentially work need to verify). It won't accomplish what I need though, as I still won't be able to provide a custom message when there's a mismatch. Similar to most unit testing framework, a reason string that's presented when assertion is failing.
